### ISSUE TITLE
Horovod PyTorch Multi-GPU Only Using 1 GPU (Multi-Node Too)

### ISSUE THREAD

**Issue by McCrearyD** [awslabs/amazon-sagemaker-examples#1246]
For the Horovod + PyTorch + MNIST example, I created my own algorithm deriving this method. From what I can tell, it seems like SageMaker is only taking advantage of 1 GPU per node even when using ml.p3.16xlarge instances. hvd.size() will always return 1 and hvd.local_rank() always returns 0. 

It probably has something to do with the `horovodrun` command under the hood. Take a look at the very bottom of this page: https://horovod.readthedocs.io/en/stable/pytorch.html

It requires extra params for launching multi GPU.

The reason I think this is the problem is if you look at [SageMaker Tensorflow Estimator's documentation](https://sagemaker.readthedocs.io/en/stable/frameworks/tensorflow/sagemaker.tensorflow.html#tensorflow-estimator), there is an arg called `distributions`. According to [this article by AWS](https://aws.amazon.com/blogs/machine-learning/launching-tensorflow-distributed-training-easily-with-horovod-or-parameter-servers-in-amazon-sagemaker/) this is where you need to define your horovod (mpi) options. More specifically `hvd_processes_per_host`.

In contrast, the [SageMaker PyTorch Estimator's documentation](https://sagemaker.readthedocs.io/en/stable/frameworks/pytorch/sagemaker.pytorch.html) has no such argument. Neither do their base classes ([Framework Estimator](https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html#sagemaker.estimator.Framework) and so on...).

I could very well be doing something wrong, but literally all I did was change a few hyperparameters and use a larger database than MNIST (ImageNet).

Another interesting side effect is that after 25% of the first epoch, GPU utilization drops from ~86% all the way to 0%, however the GPU memory usage remains the same.

**Comment by icywang86rui** [comment]
@McCrearyD 
Thanks for reporting this. We need to add the distribution arg in the pytorch estimator to run the training script with mpi like the tensorflow one here -  https://github.com/aws/sagemaker-python-sdk/blob/64f600d677872fe8656cdf25d68fc4950b2cd28f/doc/frameworks/tensorflow/using_tf.rst#training-with-horovod

**Comment by McCrearyD** [comment]
they fixed it in one of the latest versions, although the docker container hasn't been updated so it basically does nothing.

**Comment by austinmw** [comment]
@icywang86rui Any updates on adding Horovod / distribution parameter support for PyTorch estimator?

**Comment by austinmw** [comment]
I think I managed to solve this with a requirements file and a custom estimator object:
https://github.com/horovod/horovod/issues/1996#issuecomment-727078447
